src/families/cosmos/libcore-buildTransaction.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import type {
+  Transaction,
+  CoreCosmosLikeTransaction,
+  CoreCosmosGasLimitRequest,
+} from "./types";
+import type { Account } from "../../types";
+import type { Core, CoreAccount, CoreCurrency } from "../../libcore/types";
+
+import {
+  bigNumberToLibcoreAmount,
+  libcoreBigIntToBigNumber,
+} from "../../libcore/buildBigNumber";
+import { BigNumber } from "bignumber.js";
+import { cosmosCreateMessage } from "./message";
+import { getEnv } from "../../env";
+import { promiseAllBatched } from "../../promise";
+
+export async function cosmosBuildTransaction({
+  account,
+  core,
+  coreAccount,
+  coreCurrency,
+  transaction,
+  isCancelled,
+}: {
+  account: Account,
+  core: Core,
+  coreAccount: CoreAccount,
+  coreCurrency: CoreCurrency,
+  transaction: Transaction,
+  isPartial: boolean,
+  isCancelled: () => boolean,
+}): Promise<?CoreCosmosLikeTransaction> {
+  const { fees, gasLimit, memo } = transaction;
+
+  const cosmosLikeAccount = await coreAccount.asCosmosLikeAccount();
+  if (isCancelled()) return;
+
+  const transactionBuilder = await cosmosLikeAccount.buildTransaction();
+  if (isCancelled()) return;
+
+  const messages = await cosmosCreateMessage(
+    account.freshAddress,
+    transaction,
+    core
+  );
+
+  promiseAllBatched(
+    3,
+    messages,
+    async (message) => await transactionBuilder.addMessage(message)
+  );
+
+  const memoTransaction = memo ? memo : "";
+  await transactionBuilder.setMemo(memoTransaction);
+
+  // Gas
+  let gas: BigNumber;
+
+  if (gasLimit && gasLimit !== "0") {
+    gas = BigNumber(gasLimit);
+  } else {
+    const gasRequest: CoreCosmosGasLimitRequest = {
+      memo: memoTransaction,
+      amplifier: getEnv("COSMOS_GAS_AMPLIFIER"),
+      messages,
+    };
+    gas = await libcoreBigIntToBigNumber(
+      await cosmosLikeAccount.estimateGas(gasRequest)
+    );
+  }
+  const gasAmount = await bigNumberToLibcoreAmount(core, coreCurrency, gas);
+  if (isCancelled()) return;
+
+  await transactionBuilder.setGas(gasAmount);
+
+  const gasPrice = getEnv("COSMOS_GAS_PRICE");
```

Comment:
this should be part of the transaction object (as the other env)

Reply:
This is only related to the actual instance of the Gaia backend we're using, not to the transaction theoretically